estate/models/property_offer.py

```diff
@@ -72,3 +72,14 @@ def action_refuse_offer(self):
             record.status = "refused"
             if record.property_id.selling_price == record.price:
                 record.property_id.selling_price = 0
+
+    @api.model
+    def create(self, vals):
+        property_id = self.env["estate.property"].browse(vals["property_id"])
+
+        if property_id.best_price > vals["price"] or not vals["price"]:
```

Comment:
Unfortunately, this won't work. You should first check that `price` exists in `vals` (since it is not required, it may not be in the dict). And if not, calling `vals["price"]` will raise. Instead, do something like this
- Use `get` to retrieve the value if it exists
- Fallback on something that will raise (0 in this case)
```suggestion
        if property_id.best_price > vals.get('price', 0):
```